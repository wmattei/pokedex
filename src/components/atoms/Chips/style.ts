import { StyleSheet } from 'react-native';
import { Theme } from 'react-native-paper/lib/typescript/types';

export default (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      marginLeft: 10,
    },
    item: {
      margin: 4,
      height: 35,
      backgroundColor: theme.colors.accent,
    },
  });
