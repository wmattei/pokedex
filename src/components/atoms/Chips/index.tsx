import * as React from 'react';
import { ScrollView } from 'react-native-gesture-handler';
import { Chip as ChipPaper, useTheme } from 'react-native-paper';
import style from './style';

type Props = {
  chips: Array<{ label: string; id: string }>;
};

export default function Chips({ chips }: Props) {
  const theme = useTheme();

  return (
    <ScrollView horizontal style={style(theme).container}>
      {chips.map((chip) => (
        <ChipPaper
          style={style(theme).item}
          key={`chip-${chip.id || chip.label}`}
        >
          {chip.label}
        </ChipPaper>
      ))}
    </ScrollView>
  );
}
