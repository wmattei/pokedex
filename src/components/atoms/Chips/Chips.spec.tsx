import * as React from 'react';
import { create } from 'react-test-renderer';
import Chips from '.';

describe('Chips', () => {
  it('Renders correctly', () => {
    const chips = [
      { id: 'chip1', label: 'Chip1' },
      { id: 'chip2', label: 'Chip2' },
    ];
    const tree = create(<Chips chips={chips} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
