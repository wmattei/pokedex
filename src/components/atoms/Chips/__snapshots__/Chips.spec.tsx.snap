// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips Renders correctly 1`] = `
<RCTScrollView
  horizontal={true}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#03dac4",
          "borderColor": "#03dac4",
          "borderRadius": 16,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "elevation": 0,
          "flexDirection": "row",
          "height": 35,
          "margin": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
            "selected": false,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingLeft": 4,
                "position": "relative",
              },
              Object {
                "paddingRight": 4,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "marginVertical": 4,
                    "minHeight": 24,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "marginLeft": 8,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            Chip1
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#03dac4",
          "borderColor": "#03dac4",
          "borderRadius": 16,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "elevation": 0,
          "flexDirection": "row",
          "height": 35,
          "margin": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
            "selected": false,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingLeft": 4,
                "position": "relative",
              },
              Object {
                "paddingRight": 4,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "lineHeight": 24,
                    "marginVertical": 4,
                    "minHeight": 24,
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "marginLeft": 8,
                    "marginRight": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            Chip2
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
